bson/setter_getter.go

```diff
@@ -34,7 +32,7 @@ import (
 //	    return raw.Unmarshal(s)
 //	}
 type Setter interface {
-	SetBSON(raw bson.RawValue) error
+	SetBSON(raw RawValue) error
 }
```

Comment:
Moving this to the `bson` package creates ambiguous naming because the `Setter` interface is only relevant to mgo compatibility, but overlaps confusingly with the `bson.Unmarshaler` interface. Is there a reason we need to move these APIs to the `bson` package instead of keeping them in the `mgocompat` package?

Reply:
I moved these methods into the `bson` package because we un-exposed codec structs and their fields, which made them inaccessible from `mgocompat`.